packages/bar/src/BarTotals.tsx

```diff
@@ -0,0 +1,154 @@
+
+        bars.forEach(bar => {
+            const { indexValue, value } = bar.data
+            updateTotalsByIndex(totalsByIndex, indexValue, Number(value))
+            updateTotalsPositivesByIndex(totalsPositivesByIndex, indexValue, Number(value))
+        })
+
+        totalsPositivesByIndex.forEach((totalsPositive, indexValue) => {
+            let xPosition: number
+            let yPosition: number
+
+            if (layout === 'vertical') {
+                xPosition = xScale(indexValue)
+                yPosition = yScale(totalsPositive)
+            } else {
+                xPosition = xScale(totalsPositive)
+                yPosition = yScale(indexValue)
+            }
+
+            totals.push(
+                <text
+                    key={'total_' + indexValue}
+                    x={xPosition + (layout === 'vertical' ? barWidth / 2 : xOffsetHorizontally)}
+                    y={yPosition + (layout === 'vertical' ? yOffsetVertically : barHeight / 2)}
+                    {...commonProps}
+                >
+                    {totalsByIndex.get(indexValue)}
+                </text>
+            )
+        })
+    }
+
+    if (groupMode === 'grouped') {
+        const greatestValueByIndex = new Map<string | number, number>()
+        const numberOfBarsByIndex = new Map()
+
+        bars.forEach(bar => {
+            const { indexValue, value } = bar.data
+            updateTotalsByIndex(totalsByIndex, indexValue, Number(value))
+            updateGreatestValueByIndex(greatestValueByIndex, indexValue, Number(value))
+            updateNumberOfBarsByIndex(numberOfBarsByIndex, indexValue)
+        })
+
+        greatestValueByIndex.forEach((greatestValue, indexValue) => {
+            let xPosition: number
+            let yPosition: number
+
+            if (layout === 'vertical') {
+                xPosition = xScale(indexValue)
+                yPosition = yScale(greatestValue)
+            } else {
+                xPosition = xScale(greatestValue)
+                yPosition = yScale(indexValue)
+            }
+
+            const indexBarsWidth = numberOfBarsByIndex.get(indexValue) * barWidth
+            const indexBarsHeight = numberOfBarsByIndex.get(indexValue) * barHeight
+
+            totals.push(
+                <text
+                    key={'total_' + indexValue}
+                    x={
+                        xPosition +
+                        (layout === 'vertical' ? indexBarsWidth / 2 : xOffsetHorizontally)
+                    }
+                    y={
+                        yPosition +
+                        (layout === 'vertical' ? yOffsetVertically : indexBarsHeight / 2)
+                    }
+                    {...commonProps}
+                >
+                    {totalsByIndex.get(indexValue)}
+                </text>
+            )
+        })
+    }
+
+    return <>{totals}</>
+}
+
+export const updateTotalsByIndex = (
```

Comment:
If we extract the computation in a hook, we might not need all these helpers.

Reply:
I created a central hook for svg and canvas but I kept the helpers in order to document with comments and not repeating code.

Reply:
Makes sense.